Documentation/MigrationGuide.md

```diff
@@ -1,46 +1,69 @@
 
-#### Colors
+## Changes for template writers
 
-- `enumName`: has been replaced by `param.enumName`, should provide default value.
-- `colors`: replaced by the `palettes` array, each having a `name` and a `colors` property.
-- For each `color`:
-  - `rgb` and `rgba`: can be composed from the other components.
+As a reminder, you can find all the documentation for the context structures provided as variables to your templates [in the Documentation folder of this repository](https://github.com/SwiftGen/SwiftGenKit/tre/master/Documentation/) — one MarkDown file for each SwiftGen subcommand / SwiftGenKit parser.
 
-#### Fonts
+### Common changes and the new `--param` flags
 
-- `enumName`: has been replaced by `param.enumName`, should provide default value.
-- For each `font`:
-  - `fontName`: has been replaced by the `name` property.
+One common changes across all templates is that the `enumName` variable (or `sceneEnumName` & `segueEnumName` for storyboards) have been replaced by their `param.enumName` counterparts. Those are variables provided by the user via the `--param enumName=…` flag during SwiftGen's command line invocation.
 
-#### Images
+This means that you are now also responsible for providing a default value for those `param.enumName` if you use them, in case the user didn't provide the `--param enumName=…` flag at all. You can use that with Stencil's `default` filter, e.g. `enum {{param.enumName|default:"Assets"}}`
 
-- `enumName`: has been replaced by `param.enumName`, should provide default value.
-- `images`: just old, `catalogs` contains the structured information.
+You can also take advantage of that new `--param` feature to make your own templates more customizable, by allowing users to provide arbitrary values via the command line, e.g. using `{{param.foo|default:"Foo"}}` and `{{param.bar|default:"-"}}` in your templates to let users provide custom values using `--param foo=MyFoo --param bar=_`. Just don't forget to document the available params somewhere to let the users of your templates know about those.
 
-#### Storyboards
+### Colors
 
-- `extraImports`: replaced by `modules` (https://github.com/AliSoftware/SwiftGen/pull/243)
-- `sceneEnumName`: has been replaced by `param.sceneEnumName`, should provide default value.
-- `segueEnumName`: has been replaced by `param.segueEnumName`, should provide default value.
-- For each `scene`:
-  - `isBaseViewController`: removed. You can replace it with a test for `baseType == "ViewController"`.
+_📖 see the full context structure [in the documentation here](https://github.com/SwiftGen/SwiftGenKit/blob/master/Documentation/Colors.md)._
 
-#### Strings
+- `enumName`: This variable has been replaced by `param.enumName` — see above.
```

Comment:
Maybe we could set a link on all the `see above` to the designated section `#common-changes-and-the-new---param-flags`?